<?xml version="1.0"?>
<launch>
	<arg name="sim" default="false"/>

	<node pkg="usb_cam" type="usb_cam_node" name="top_front_camera" unless="$(arg sim)">
		<param name="video_device" value="/dev/video3"/>
		<param name="camera_name" value="top_front_camera"/>
		<param name="camera_frame_id" value="top_front_camera_optical_frame"/>
		<param name="image_width" value="640"/>
		<param name="image_height" value="480"/>
		<param name="pixel_format" value="yuyv" />
	</node>

	<group ns="top_front_camera">
		<node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager"/>

		<node pkg="image_transport" type="republish" name="decompress" args="compressed in:=image_raw raw out:=image_raw" if="$(arg sim)"/>

		<!--camera_nodelet_manager-->
		<node pkg="nodelet" type="nodelet" name="rectify" args="load image_proc/rectify camera_nodelet_manager">
			<remap from="image_mono" to="image_raw"/>
			<remap from="camera_info" to="camera_info"/>
			<remap from="image_rect" to="image_rect"/>
			<param name="queue_size" value="5"/>
		</node>
		<!-- qhd -->
		<node pkg="nodelet" type="nodelet" name="qhd_resize" args="load image_proc/resize camera_nodelet_manager">
			<remap from="image" to="image_raw"/>
			<remap from="camera_info" to="camera_info"/>
			<remap from="~image" to="qhd/image_raw"/>
			<remap from="~camera_info" to="qhd/camera_info"/>
			<param name="use_scale" value="1"/>
			<param name="scale_width" value="0.5"/>
			<param name="scale_height" value="0.5"/>
			<param name="queue_size" value="5"/>
		</node>
		<node pkg="nodelet" type="nodelet" name="qhd_rectify" args="load image_proc/rectify camera_nodelet_manager">
			<remap from="image_mono" to="qhd/image_raw"/>
			<remap from="camera_info" to="qhd/camera_info"/>
			<remap from="image_rect" to="qhd/image_rect"/>
			<param name="queue_size" value="5"/>
		</node>

		<!-- sd -->
		<node pkg="nodelet" type="nodelet" name="sd_resize" args="load image_proc/resize camera_nodelet_manager">
			<remap from="image" to="image_raw"/>
			<remap from="camera_info" to="camera_info"/>
			<remap from="~image" to="sd/image_raw"/>
			<remap from="~camera_info" to="sd/camera_info"/>
			<param name="use_scale" value="1"/>
			<param name="scale_width" value="0.25"/>
			<param name="scale_height" value="0.25"/>
			<param name="queue_size" value="5"/>
		</node>
		<node pkg="nodelet" type="nodelet" name="sd_rectify" args="load image_proc/rectify camera_nodelet_manager">
			<remap from="image_mono" to="sd/image_raw"/>
			<remap from="camera_info" to="sd/camera_info"/>
			<remap from="image_rect" to="sd/image_rect"/>
			<param name="queue_size" value="5"/>
		</node>
	</group>
</launch>
